File: security/src/main/java/top/bettercode/summer/security/ClientAuthorize.java
package top.bettercode.summer.security;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * client 权限标识
 *
 * @author Peter Wu
 */
@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Inherited
@Documented
public @interface ClientAuthorize {

}

File: web/src/main/kotlin/top/bettercode/summer/security/authorize/ClientAuthorize.kt
package top.bettercode.summer.security.authorize

import java.lang.annotation.Inherited

/**
 * client 权限标识
 *
 * @author Peter Wu
 */
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.CLASS
)
@Retention(AnnotationRetention.RUNTIME)
@Inherited
@MustBeDocumented
annotation class ClientAuthorize(
    val scope: Array<String> = [],
)
